### ISSUE TITLE
Support for PhoenixGo

### ISSUE THREAD

**Issue by pnprog** [pnprog/goreviewpartner#99]
Hi @wonderingabout ,

I'm opening an issue here to discuss about the bug in [issue 86](https://github.com/Tencent/PhoenixGo/issues/86) to avoid polluting PhoenixGo.

Looking at the RSGF file you share, I can notice that only three variations are provided (width=3), and they are always 1 move deep (depth=1) => this is probably another issue, but just to be sure, can you double check the values **inprint_tree_depth** and **print_tree_width** in **mcts_something.conf** ?

On my own computer, I got 6 variations (most of the time) and most of them are 2 or 3 moves deep, with a very small number or playout. Although, the win/rate graph is quite similar to that of LeelaZero with PhoenixGo weights. So this is definitely a parsing issue. Maybe the output is a bit different when using GPU or CPU?

So, I need to see by myself what is the output of PhoenixGo from your computer. Here is what I propose:
 + change the values of  **inprint_tree_depth** and **print_tree_width** at 3 and 3, in **mcts_something.conf** (just to reduce the quantity of log, maybe you can reduce the thinking time to 20s also)
+ In GRP => Settings => PhoenixGo, open a test windows, and send the GTP command **genmove black** then copy here the whole log

Hopefully, I can find something :)

**Comment by pnprog** [comment]
Maybe I found the problem: it could be related to pondering.
Can you disable pondering and restart the analysis again?
In the mctf conf file:
`enable_background_search: 0 `
Apparently, when pondering is on, when `clear_board` GTP instruction is used, PhoenixGo dumps the current result of the tree search, and those extra data (formatted as the other data) is confusing GRP.

**Comment by pnprog** [comment]
Yes, I can confirm that my GRP has similar symptoms when pondering is ON.
At least, this is something easy to fix.

**Comment by wonderingabout** [comment]
i was thinking about pondering too, but i thought maybe phoenixgo is smart

i used 6x6 in .conf file of phoenixgo

but on grp settings i used max variations 3, with 25 depth

i think it was fine to pollute the phoenixgo github, after all its related

i'll link you my conf file when i finish updating the PR

**Comment by pnprog** [comment]
> i think it was fine to pollute the phoenixgo github, after all its related

That probably ok, but I don't want to have my bugs advertised all over github as well :P

I just pushed a commit that should help with this pondering issue. But try first without pondering, to see if everything else is OK. I will be running a test with pondering during the night.

**Comment by wonderingabout** [comment]
then i'll retry with same version and ponderingoff this time, same as this file : 

https://github.com/wonderingabout/PhoenixGo/blob/faqv2-bazel-master/etc/mcts_1gpu_grp.conf

**Comment by wonderingabout** [comment]
oh wow @pnprog  !!!

its 1000 times better with pondering off lol

http://www.mediafire.com/file/vcenr3jzq2vaxqs/game2v3.rsgf

native phoenixgo 3200 simulations per move : 

![graph3](https://user-images.githubusercontent.com/38690718/51790901-70586700-219b-11e9-8f7b-43772bc70aa3.png)

compare it with LZ-phoenixgo-orig -v 3200 : 

![lzphoenixgogrp](https://user-images.githubusercontent.com/38690718/51790910-9bdb5180-219b-11e9-9cbb-f19fe1ee33b0.png)

**Comment by wonderingabout** [comment]
@alreadydone

**Comment by wonderingabout** [comment]
compare it to lz-phoenixgo-orig : https://github.com/Tencent/PhoenixGo/issues/86#issuecomment-456697268

**Comment by wonderingabout** [comment]
now i want to test width 3 + depth 15

(i think seeing 3 alternate moves is reasonable + with enough diversity + they will be the strongest)

**Comment by wonderingabout** [comment]
result v4 :+1: 

http://www.mediafire.com/file/83ajv4mytdw3ld3/game2v4.rsgf

![graph4](https://user-images.githubusercontent.com/38690718/51792673-f7660900-21b4-11e9-9181-e4cc9aaa67ff.png)

comapre to lz-phoenixgo-orig :

**Comment by pnprog** [comment]
Good to see it's starts to be ok :)

> i think seeing 3 alternate moves is reasonable + with enough diversity + they will be meaningful enough (enough simulations spent on them)

What can be done is to filter the variations to display (and among the variations, the number of moves) using the number of playouts done. I did something similar with the implementation for Pachi.

**Comment by wonderingabout** [comment]
isnt it what is already done here ?

when i set variations to display to 3 ?

**Comment by wonderingabout** [comment]
btw i sent you this yesterday @pnprog :+1: 

https://github.com/pnprog/goreviewpartner/pull/100

i think it will help clear away many of the confusion :)

**Comment by wonderingabout** [comment]
so, i'm closing the PhoenixGo github issues in : 
https://github.com/Tencent/PhoenixGo/issues/86

to avoid duplicates,

let's continue here whenever we can : 
i hope that i will be ready when we can continue this :+1: (but it's working quite good already)

**Comment by pnprog** [comment]
Hi @wonderingabout ,

I see your PR for PhoenixGo have been accepted. Can you update/push again your PR for GRP? (I made some changes in config.ini since then, so the PR cannot be merged as it is)

**Comment by wonderingabout** [comment]
hi too @pnprog :+1: 

ok, let me look
